src/pymmcore_widgets/useq_widgets/points_plans/_well_graphics_view.py

```diff
@@ -70,40 +72,85 @@ def setPointsPlan(self, plan: useq.RelativeMultiPointPlan) -> None:
         if hasattr(plan, "shape") and isinstance(plan.shape, Shape):
             self._is_circular = plan.shape == Shape.ELLIPSE
 
-        self._draw_outline()
+        # FOV CONSTRAIN AREA
+        if isinstance(plan, useq.RandomPoints):
+            self._draw_fov_constrain_area(plan.max_width, plan.max_height)
+        elif self._fov_constrain_area_item:
+            self._scene.removeItem(self._fov_constrain_area_item)
+
+        # WELL OUTLINE
+        self._draw_well_outline()
+
+        # DRAW FOVS
         self._draw_fovs(plan)
 
-    def _draw_outline(self) -> None:
+    def _draw_well_outline(self) -> None:
         """Draw the outline of the well area."""
-        if self._outline_item:
-            self._scene.removeItem(self._outline_item)
+        if self._well_outline_item:
+            self._scene.removeItem(self._well_outline_item)
         if (rect := self._well_rect()).isNull():
             return
 
         pen = QPen(QColor(Qt.GlobalColor.green))
         pen.setWidth(self._scaled_pen_size())
         if self._is_circular:
-            self._outline_item = self._scene.addEllipse(rect, pen=pen)
+            self._well_outline_item = self._scene.addEllipse(rect, pen=pen)
         else:
-            self._outline_item = self._scene.addRect(rect, pen=pen)
+            self._well_outline_item = self._scene.addRect(rect, pen=pen)
         self._resize_to_fit()
 
+    def _draw_fov_constrain_area(self, width: float, height: float) -> None:
+        """Draw the bounding area to constrain the FOVs."""
+        if self._fov_constrain_area_item:
+            self._scene.removeItem(self._fov_constrain_area_item)
+
+        if self._well_width_um is None or self._well_height_um is None:
+            return
+
+        # bounding area to constrain the FOVs
+        rect = (
+            QRectF(-width / 2, -height / 2, width, height)
+            if self._is_circular
+            else QRectF(-(width - (width / 2)), -(height - (height / 2)), width, height)
+        )
+
+        pen = QPen(QColor(Qt.GlobalColor.green))
+        pen.setWidth(self._scaled_pen_size())
+        pen.setStyle(Qt.PenStyle.DotLine)
+
+        if self._is_circular:
```

Comment:
technically the bounding area should come from the `RandomPoints.shape`, where `self._is_circular` should come from `WellPlate.circular_wells`.  i.e. there's nothing conceptually wrong with constraining the points into a square bounding area inside a circular well

Reply:
so do you want to add a `well_plate: WellPlate | None = None` arg to the `PointsPlanWidget`?

Reply:
or maybe something more general like `well_info: useq.WellPlate | tuple[float, float] | float | None = None` which can include either a `WellPlate` or simply a `well_size`?

Reply:
i don't think it needs awareness, or a pointer to, a full plate object.  It just needs to track and retain the difference in meaning between the shape of the "world" in which this single well lives (well size and shape) and the "bounds" in which the points live (bounds size and shape).  when used in the *context* of some larger widget that manages plates and stuff, that outer widget might choose to set the well size/shape of this widget accordingly, but this widget itself doesn't need to know anything about a full well plate